staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_test.go

```diff
@@ -79,85 +80,85 @@ func checkStorageInvariants(ctx context.Context, t *testing.T, key string) {
 }
 
 func TestCreate(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestCreate(ctx, t, cacher, checkStorageInvariants)
 }
 
 func TestCreateWithTTL(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestCreateWithTTL(ctx, t, cacher)
 }
 
 func TestCreateWithKeyExist(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestCreateWithKeyExist(ctx, t, cacher)
 }
 
 func TestGet(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestGet(ctx, t, cacher)
 }
 
 func TestUnconditionalDelete(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestUnconditionalDelete(ctx, t, cacher)
 }
 
 func TestConditionalDelete(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestConditionalDelete(ctx, t, cacher)
 }
 
 func TestDeleteWithSuggestion(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestDeleteWithSuggestion(ctx, t, cacher)
 }
 
 func TestDeleteWithSuggestionAndConflict(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestDeleteWithSuggestionAndConflict(ctx, t, cacher)
 }
 
 func TestDeleteWithSuggestionOfDeletedObject(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestDeleteWithSuggestionOfDeletedObject(ctx, t, cacher)
 }
 
 func TestValidateDeletionWithSuggestion(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestValidateDeletionWithSuggestion(ctx, t, cacher)
 }
 
 func TestPreconditionalDeleteWithSuggestion(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, _, terminate := testSetup(t)
 	t.Cleanup(terminate)
 	storagetesting.RunTestPreconditionalDeleteWithSuggestion(ctx, t, cacher)
 }
 
 func TestList(t *testing.T) {
-	ctx, cacher, terminate := testSetup(t)
+	ctx, cacher, server, terminate := testSetup(t)
 	t.Cleanup(terminate)
-	storagetesting.RunTestList(ctx, t, cacher, true)
+	storagetesting.RunTestList(ctx, t, cacher, etcd3.CompactTestStorage(server.V3Client), true)
```

Comment:
Instead of compacting just etcd3 storage, please pipe it into:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_testing_utils_test.go#L117

Reply:
Ok, but I don't really understand the logic of compacting watch cache. I mean the dependency reversion doesn't make sense for me in this function. Normally it's not that watch cache compacts storage, storage compacts itself and watch cache then can notice it.

I updated the code, but I didn't dig deep so I'm not sure if it is correct